lib/ipam/ipam.go

```diff
@@ -226,10 +227,22 @@ func (c ipamClient) determinePools(requestedPoolNets []net.IPNet, version int) (
 
 		// Make sure each requested pool exists
 		for _, rp := range requestedPoolNets {
```

Comment:
So I think there's an issue here, in that this loop only gets executed if `requestedPoolNets > 0`.

Otherwise, we just use all enabled IP pools. I think we need to rework that in order to meet the desired behavior.

If no requestedPoolNets are provided, we still want to filter out only the pools which select this node.

Reply:
addressed 9fe3116